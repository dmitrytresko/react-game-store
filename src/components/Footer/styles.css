.footer {
  position: fixed;
  bottom: 0;
  z-index: 10;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  padding: 15px 0;
  background-color: black;
}

.footer__tagline {
  margin-bottom: 15px;
  color: white;
  font-size: 1.45rem;
  font-family: "Teko", sans-serif;
  text-decoration: none;
}

.footer__icons-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 25%;
  min-width: 350px;
}

.footer__link {
  text-decoration: none;
  cursor: pointer;
}

.footer__img {
  width: 35px;
  height: 35px;
}

.footer__img--medium {
  width: 50px;
  height: 35px;
}

.footer__img--large {
  width: 70px;
  height: 35px;
}
