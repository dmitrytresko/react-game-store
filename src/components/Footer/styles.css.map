{
    "version": 3,
    "mappings": "AAAA,AAAA,OAAO,CAAC;EACN,QAAQ,EAAE,KAAK;EACf,MAAM,EAAE,CAAC;EACT,OAAO,EAAE,EAAE;EACX,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;EACvB,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,MAAM;EACf,gBAAgB,EAAE,KAAK;CAoCxB;;AAlCE,AAAD,gBAAU,CAAC;EACT,aAAa,EAAE,IAAI;EACnB,KAAK,EAAE,KAAK;EACZ,SAAS,EAAE,OAAO;EAClB,WAAW,EAAE,kBAAkB;EAC/B,eAAe,EAAE,IAAI;CACtB;;AAEA,AAAD,wBAAkB,CAAC;EACjB,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,aAAa;EAC9B,KAAK,EAAE,GAAG;EACV,SAAS,EAAE,KAAK;CACjB;;AAEA,AAAD,aAAO,CAAC;EACN,eAAe,EAAE,IAAI;EACrB,MAAM,EAAE,OAAO;CAChB;;AAEA,AAAD,YAAM,CAAC;EACL,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;CAWb;;AATE,AAAD,oBAAS,CAAC;EACR,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;CACb;;AAEA,AAAD,mBAAQ,CAAC;EACP,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,IAAI;CACb",
    "sources": [
        "styles.scss"
    ],
    "names": [],
    "file": "styles.css"
}