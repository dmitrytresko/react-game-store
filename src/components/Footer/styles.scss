$tabletMediaWidth: 768px;

.footer {
  bottom: 0;
  z-index: 10;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 125px;
  margin-top: 20px;

  padding: 15px 0;
  background-color: black;

  &__tagline {
    margin-bottom: 15px;
    color: white;
    font-size: 1.45rem;
    font-family: "Teko", sans-serif;
    text-decoration: none;

    @media (max-width: ($tabletMediaWidth - 1px)) {
      font-size: 1.25rem;
    }
  }

  &__icons-container {
    display: flex;
    justify-content: space-between;
    width: 25%;
    min-width: 300px;
  }

  &__link {
    text-decoration: none;
    cursor: pointer;
  }

  &__img {
    width: 40px;
    height: 40px;

    @media (max-width: $tabletMediaWidth) {
      width: 25px;
      height: 25px;
    }

    &--medium {
      width: 55px;
      height: 40px;

      @media (max-width: $tabletMediaWidth) {
        width: 40px;
        height: 25px;
      }
    }

    &--large {
      width: 80px;
      height: 40px;

      @media (max-width: $tabletMediaWidth) {
        width: 50px;
        height: 25px;
      }
    }
  }
}
