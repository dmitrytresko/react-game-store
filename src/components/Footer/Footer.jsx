import blizzardImg from "../../assets/img/blizzard.png";
import eaImg from "../../assets/img/electronic_arts.png";
import ubisoftImg from "../../assets/img/ubisoft.png";
import rockstarImg from "../../assets/img/rockstar.png";
import valveImg from "../../assets/img/valve.png";
import "./styles.scss";

const Footer = () => {
  return (
    <footer className="footer">
      <p className="footer__tagline">Incredible convenient</p>
        <div className="footer__icons-container">
          <a href="https://www.blizzard.com/en-us/" className="footer__link" rel="noreferrer" target="_blank">
            <img className="footer__img--large" src={blizzardImg} alt="Blizzard logo"></img>
          </a>
          <a href="https://www.ea.com/sports" className="footer__link" rel="noreferrer" target="_blank">
            <img className="footer__img" src={eaImg} alt="EA logo"></img>
          </a>
          <a href="https://www.ubisoft.com/en-us/" className="footer__link" rel="noreferrer" target="_blank">
            <img className="footer__img--medium" src={ubisoftImg} alt="Ubisoft logo"></img>
          </a>
          <a href="https://www.rockstargames.com/" className="footer__link" rel="noreferrer" target="_blank">
            <img className="footer__img" src={rockstarImg} alt="Rockstar logo"></img>
          </a>
          <a href="https://www.valvesoftware.com/en/" className="footer__link" rel="noreferrer" target="_blank">
            <img className="footer__img--large" src={valveImg} alt="Valve logo"></img>
          </a>
         </div>
    </footer>
  )
}

export default Footer;
