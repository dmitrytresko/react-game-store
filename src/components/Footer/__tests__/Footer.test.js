import React from "react";
import { render } from "@testing-library/react";
import Footer from "../Footer";

describe("Footer", () => {
  it("should render Footer component", () => {
    const { container, getByText, getAllByRole } = render(<Footer />);

    const links = container.getElementsByClassName('footer__link');
    const newLinks = getAllByRole('link');

    expect(links.length).toEqual(5);
    expect(newLinks.length).toEqual(5);
    expect(getByText('Incredible convenient')).toBeTruthy();
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})
