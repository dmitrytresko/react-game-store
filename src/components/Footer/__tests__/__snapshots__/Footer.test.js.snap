// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render Footer component 1`] = `
<div>
  <footer
    class="footer"
  >
    <p
      class="footer__tagline"
    >
      Incredible convenient
    </p>
    <div
      class="footer__icons-container"
    >
      <a
        class="footer__link"
        href="https://www.blizzard.com/en-us/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Blizzard logo"
          class="footer__img--large"
          src="test-file-stub"
        />
      </a>
      <a
        class="footer__link"
        href="https://www.ea.com/sports"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="EA logo"
          class="footer__img"
          src="test-file-stub"
        />
      </a>
      <a
        class="footer__link"
        href="https://www.ubisoft.com/en-us/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Ubisoft logo"
          class="footer__img--medium"
          src="test-file-stub"
        />
      </a>
      <a
        class="footer__link"
        href="https://www.rockstargames.com/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Rockstar logo"
          class="footer__img"
          src="test-file-stub"
        />
      </a>
      <a
        class="footer__link"
        href="https://www.valvesoftware.com/en/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Valve logo"
          class="footer__img--large"
          src="test-file-stub"
        />
      </a>
    </div>
  </footer>
</div>
`;
